src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -392,19 +407,13 @@ public static GapicProductConfig createDummyInstance(
         configSchemaVersion);
   }
 
-  private static ImmutableMap<String, InterfaceConfig> createInterfaceConfigMap(
+  /** Return the list of information about clients to be generated. */
+  private static ImmutableList<GapicInterfaceInput> createInterfaceInputs(
```

Comment:
If you agree with my comment below about the `createInterfaceInput` method, then perhaps also this should be two methods? The only shared code is the construction of the `protoInterfaces` map - but that could be pulled out, and passed into each method instead of sourceProtos.

Reply:
Hmm, so at this point in the refactoring, the if/else block on `gapicConfigPresent` is one method call per each branch, so there's like 60 LOC that's shared, and only 1 that differs depending on the value of the flag. I'm wary of extracting out the if/else block to the parent caling method because the if/else block is inside a medium-size for-loop, and then that for-loop would have to be moved into it's calling method `GapicProduct.create()` which is already very long.

Reply:
Fair enough, although I would argue that lines 436-451, and all of the lines inside the for loop that reference `interfaceConfigProto`, are not really "shared", since they only affect one branch of the if statement. So I wouldn't pull the for loop out, I would create a `createInterfaceInputsWithConfigProto` method that handled the case with gapicConfigPresent==true, and modify `createInterfaceInputs` to handle only the false case, and have the shared `protoInterfaces` variable be an input instead of `sourceProtos`. But I also feel like this is less important that other comments, so feel free to disregard if you disagree.

Reply:
(or, keep passing in `sourceProtos` but move creation of `protoInterfaces` variable into its own method and share it between the two methods)

Reply:
Ok, moved the switching logic into GapicProductConfig.create(), so we can remove the `configPresence` flag altogether, assuming an empty GAPIC config is the same thing as a null one.